src/tdymemory.c

```diff
@@ -178,6 +178,22 @@ PetscErrorCode TDyAllocate_RealArray_4D(PetscReal *****array_4D, PetscInt ndim_1
   PetscFunctionReturn(0);
 }
 
+/* ---------------------------------------------------------------- */
+PetscErrorCode TDyDeallocate_RealArray_1D(PetscReal *array_1D) {
+
+  PetscFunctionBegin;
+  free(array_1D);
```

Comment:
We seem to be using `malloc` and `free` in some places and `PetscMalloc` and `PetscFree` in others. Should we decide on one or the other? If so, I'm happy to switch us over to our choice (in this PR or in another).